numba/tests/test_linalg.py

```diff
@@ -720,5 +724,116 @@ def check(a, **kwargs):
                                   (np.array([[1., 2., ], [np.inf, np.nan]],
                                             dtype=np.float64),))
 
+
+class TestLinalgQr(TestLinalgBase):
+    """
+    Tests for np.linalg.qr.
+    """
+
+    def sample_matrix(self, size, dtype, order):
+        break_at = 10
+        jmp = 0
+        # have a few attempts at shuffling to get the condition number down
+        # else not worry about it
+        mn = size[0] * size[1]
+        np.random.seed(0)  # repeatable seed
+        while jmp < break_at:
```

Comment:
Yeah, if the same function is reused it should be factored out.
As for the default return, if we don't know what to return we should raise an error instead.